wandb/apis/public/artifacts.py

```diff
@@ -501,8 +503,28 @@ def delete(self):
             )
         self.client.execute(mutation, variable_values={"id": self.id})
 
+    @property
+    def description(self):
+        """A description of the artifact collection."""
+        return self._description
+
+    @property
+    def tags(self):
+        """The tags associated with the artifact collection."""
+        return self._tags
+
+    @property
+    def name(self):
+        """The name of the artifact collection."""
+        return self._name
+
+    @property
+    def type(self):
+        """The type of the artifact collection."""
+        return self._type
+
     def __repr__(self):
-        return f"<ArtifactCollection {self.name} ({self.type})>"
+        return f"<ArtifactCollection {self._name} ({self._type})>"
```

Comment:
question: Was wondering, since I may not have enough context -- would it be worth keeping this unchanged, i.e. keep the `repr`-ed attributes as `self.name` and `self.type`?  (e.g. for continuity/consistency)

Reply:
do you mean to use the saved versions in the representation? I think what we have right now should actually be consistent with the previous implementation, since you could still rewrite `name` and `type` locally, and the representation would use those rewritten versions

Reply:
> I think what we have right now should actually be consistent with the previous implementation

definitely, agreed -- didn't meant to suggest this wasn't consistent w/previous implementation.  Mainly I was wondering about keeping the implementation unchanged here (in case of future changes):
```python
    def __repr__(self):
        return f"<ArtifactCollection {self.name} ({self.type})>"
```

so that even if `self.type` ever changes in the future (or is affected by a bug) -- if a user ever inspects or logs an `ArtifactCollection` (e.g. `logger.info(f"Artifact collection type {art_coll.type} for: {repr(art_coll)}")`), they're still guaranteed to see consistent info whether they print:
- `repr(art_collection)` / `f"{art_collection!r}"`
vs.
- `art_collection.name` and `art_collection.type`

But to echo https://github.com/wandb/wandb/pull/7528#discussion_r1592900429, I think I can see an argument both ways